src/app/calculator/compressed-air/air-leak/air-leak.service.ts

```diff
@@ -80,9 +80,17 @@ export class AirLeakService {
     this.generateExample.next(false);
   }
 
-  deleteLeak(index: number) {
-    this.airLeakInput.value.compressedAirLeakSurveyInputVec.splice(index, 1);
-    this.airLeakOutput.value.leakResults.splice(index, 1);
+  deleteLeak(index: number, settings?: Settings) {
```

Comment:
We'd normally disable the delete button if one item remains, but this calculator has an odd setup so lets stick with your solution.

- Method is only called from one place so settings doesn't need to be an optional param.
- If it's the last remaining leak, we're deleting the last leak from existing inputs and also reinitializing empty inputs. Should only need to reInit empty inputs. Take a look at what that method does if you haven't
- I'm not sure we need to resetData here. Changes should populate from initDefaultEmptyInputs.